### PR TITLE
Add Dockerfile.Debug for debug on K8S pod

### PR DESCRIPTION
**Pull request by tzheleznyak**
Explains how using delve you can debug on k8s pod

### Connected Issue/Story

Resolves #[relevant GitHub issue(s), e.g. 76]

CyberArk internal issue link: [insert issue ID](ONYX-14401)

### Definition of Done
*At least 1 todo must be completed in the sections below for the PR to be
merged.*

#### Changelog

- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [ ] This PR includes new unit and integration tests to go with the code
  changes, or
- [x] The changes in this PR do not require tests

#### Documentation

- [x] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]()
- [ ] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [ ] These changes are part of a larger initiative that will be reviewed later, or
- [x] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [ ] These changes are part of a larger initiative with a separate security review, or
- [x] There are no security aspects to these changes

### CODE REVIEW

**Review by sashaCher — COMMENTED** [review]
Some comments

**Review by sashaCher — APPROVED** [review]
Docker file is much more clear now :-)

**Review by sashaCher — APPROVED** [review]
LGTM

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -21,6 +31,121 @@ For general contribution and community guidelines, please see the [community rep
...
+2. Create kind cluster
+   `kind create cluster --config dev/kind.yaml`
+
+2. Build docker image for debugging with the `Dockerfile.Debug` running:
+   `docker build -f dev/Dockerfile.debug -t authn-k8s-client-debug:latest .`
+
+2. Load the docker image to your kind cluster
+   `kind load docker-image authn-k8s-client-debug:latest`
+
+7. Create and run a deployment with authn-k8s-client-debug:latest and your configuration. In your deployment please note this things (you may use this example)
```
Comment (szh) [inline CONTRIBUTING.md:67]: `this things` should be `these things`, or maybe just `the following:`

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -21,6 +31,106 @@ For general contribution and community guidelines, please see the [community rep
...
+               medium: Memory
+           - name: test-volume
+             hostPath:
+               path: /src/authn-k8s-client # For Kind Volume
+   ```
+
+8. Create port forwarding to the debug port in your deployment (in this example its 40000)
+   `kubectl port-forward deployment/debug-conjur-authn-k8s-client 40000:40000`
+
+9. Now you can use your favorite GO idae to configure remote debug configuration for the port you chosen and debug on your local pod.
```
Comment (szh) [inline CONTRIBUTING.md:122]: `GO idae` should be `Go IDE`

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -21,6 +31,106 @@ For general contribution and community guidelines, please see the [community rep
...
+
+2. Build docker image for debugging with the `Dockerfile.Debug` running:
+   `docker build -f dev/Dockerfile.debug -t authn-k8s-client-debug:latest .`
+
+2. Load the docker image to your kind cluster
+   `kind load docker-image authn-k8s-client-debug:latest`
+
+7. Create and run a deployment with authn-k8s-client-debug:latest and your configuration. In your deployment please note these things (you may use this example)
+
+   1. Port for debugger (In my example its 40000)
```
Comment (szh) [inline CONTRIBUTING.md:69]: (in this example it's 40000)

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -21,6 +31,106 @@ For general contribution and community guidelines, please see the [community rep
...
+
+   2. volumeMounts and volume for
```
Comment (szh) [inline CONTRIBUTING.md:71]: Not sure what this is trying to say. Maybe just `volumeMounts and volumes`?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -21,6 +31,106 @@ For general contribution and community guidelines, please see the [community rep
...
+         volumes:
+           - name: conjur-access-token
+             emptyDir:
+               medium: Memory
+           - name: test-volume
+             hostPath:
+               path: /src/authn-k8s-client # For Kind Volume
+   ```
+
+8. Create port forwarding to the debug port in your deployment (in this example its 40000)
```
Comment (szh) [inline CONTRIBUTING.md:119]: it's

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -21,6 +31,106 @@ For general contribution and community guidelines, please see the [community rep
...
+   `kubectl port-forward deployment/debug-conjur-authn-k8s-client 40000:40000`
+
+9. Now you can use your favorite GO IDE to configure remote debug configuration for the port you chosen and debug on your local pod. 
+
+   * On GoLand you can create `Go Remote` running configuration and use it to connect to delve with the local port and
```
Comment (szh) [inline CONTRIBUTING.md:124]: _In_  Goland you can create _a_ ...

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -21,6 +31,106 @@ For general contribution and community guidelines, please see the [community rep
...
+               path: /src/authn-k8s-client # For Kind Volume
+   ```
+
+8. Create port forwarding to the debug port in your deployment (in this example its 40000)
+   `kubectl port-forward deployment/debug-conjur-authn-k8s-client 40000:40000`
+
+9. Now you can use your favorite GO IDE to configure remote debug configuration for the port you chosen and debug on your local pod. 
+
+   * On GoLand you can create `Go Remote` running configuration and use it to connect to delve with the local port and 
+     debug like and other program but now on your pod. There is a debug configuration for GoLand in this repo `Debug_Authn_K8S_Client_On_Pod.xml`
```
Comment (szh) [inline CONTRIBUTING.md:125]: debug like _any_... in this repo called ...

**Review by szh — CHANGES_REQUESTED** [review]
Just a few more grammar typos, otherwise looks fantastic. I'm really excited about this, thanks for taking this on!

File: CONTRIBUTING.md (hunk 4 above)
Comment (tzheleznyak) [reply CONTRIBUTING.md:71]: volumeMounts

**Review by szh — APPROVED** [review]

**Pull request merged**